store/database.go

```diff
@@ -0,0 +1,78 @@
+package store
+
+import (
+	"io"
+)
+
+// VersionedReader extends the Reader interface by making all reads versioned.
```

Comment:
I'm confused by the word "extends", these are separate signatures from what's in `Reader`.

Reply:
"builds upon". Feel free to make a code suggestion with different nomenclature 👍